pkg/common/common.go

```diff
@@ -239,3 +241,56 @@ func GetImagePullPolicy(image string) (corev1.PullPolicy, error) {
 	}
 	return corev1.PullAlways, nil
 }
+
+// GenerateCliArgsFromConfigMap generates CLI arguments from a ConfigMap.
+//
+// This function takes a ConfigMap and a CLI subcommand(s), and returns a slice of strings representing
+// the CLI arguments from the subcommand(s) followed by the arguments from the ConfigMap.
+// The function processes each key-value pair in the ConfigMap as follows:
+//
+//  1. If the ConfigMaps' key starts with single '-' or double '--', it is left unchanged.
+//  2. If an argument key is passed without `-` or `--` prefix, then `--` is added as a key prefix.
+//  3. If the ConfigMap value is "true" or "false" (case-insensitive), it is converted to lowercase
+//     and used without single quotes surroundings (boolean value).
+//  4. The formatted key-value pair is added to the result that is alphabetically sorted.
+//
+// Args:
+//
+//		configMap: A pointer to a corev1.ConfigMap containing key-value pairs.
+//		cliSubCommand: The CLI subcommand(s) as a string, for example 'server'
+//	                or 'node-agent', 'server'
+//
+// Returns:
+//
+//	A slice of strings representing the CLI arguments.
+func GenerateCliArgsFromConfigMap(configMap *corev1.ConfigMap, cliSubCommand ...string) []string {
+
+	// cliSubCommand elements to be a separate items in args slice
+	args := make([]string, len(cliSubCommand))
+	copy(args, cliSubCommand)
```

Comment:
I think this isn't necessary. cliSubCommand should already be `[]string`. You can append to and return cliSubCommand param directly.

Reply:
@kaovilai do you want me to change this or should we agree that immutability is also important here and leave as is ?

Reply:
Immutability is important, I'm saying, you shouldn't need to make a new copy here as the cliSubcommand param is already a new copy. **The only exception is if the argument is followed by ...**, then **it is passed unchanged**.

[ref](https://go.dev/ref/spec#Passing_arguments_to_..._parameters)

> Passing arguments to ... parameters
> If f is [variadic](https://go.dev/ref/spec#Function_types) with a final parameter p of type ...T, then within f the type of p is equivalent to type []T. If f is invoked with no actual arguments for p, the value passed to p is nil. Otherwise, the value passed is a new slice of type []T with a new underlying array whose successive elements are the actual arguments, which all must be [assignable](https://go.dev/ref/spec#Assignability) to T. The length and capacity of the slice is therefore the number of arguments bound to p and may differ for each call site.
> 
> Given the function and calls
> 
> func Greeting(prefix string, who ...string)
> Greeting("nobody")
> Greeting("hello:", "Joe", "Anna", "Eileen")
> within Greeting, who will have the value nil in the first call, and []string{"Joe", "Anna", "Eileen"} in the second.
> 
> If the final argument is assignable to a slice type []T and is followed by ..., it is passed unchanged as the value for a ...T parameter. In this case no new slice is created.
> 
> Given the slice s and call
> 
> s := []string{"James", "Jasmine"}
> Greeting("goodbye:", s...)
> within Greeting, who will have the same value as s with the same underlying array.

Reply:
More reading material
https://stackoverflow.com/a/47296376/4590470

Reply:
@kaovilai updated and removed copy()